whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RemoteAddressCustomizer.java

```diff
@@ -4,43 +4,104 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.Iterables;
 import com.google.common.net.HttpHeaders;
+import com.google.common.net.InetAddresses;
+import net.ripe.db.whois.common.ip.Interval;
+import net.ripe.db.whois.common.ip.IpInterval;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jetty.http.BadMessageException;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.Request;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
 import java.net.InetSocketAddress;
+import java.util.Arrays;
 import java.util.Enumeration;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * When HTTP requests are proxied via a loadbalancer, the X-Forwarded-For value will replace the request address.
- * In case of multiple (comma separated) values in X-Forwarded-For, the last value is used.
+ * When HTTP requests via trusted source use clientIp query paramater as remote Address if set
  */
-public class RemoteAddressCustomizer  implements HttpConfiguration.Customizer {
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteAddressCustomizer.class);
+public class RemoteAddressCustomizer implements HttpConfiguration.Customizer {
 
     private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
+    public static final String QUERY_PARAM_CLIENT_IP = "clientIp";
+    private final Set<Interval> trusted;
+
+    // if client address is set in X-Forwarded-For header by HTTP proxy
+    private final boolean usingForwardedForHeader;
+
+    public RemoteAddressCustomizer(final String trustedIpRanges, final boolean xForwardedFor) {
+        this.trusted = getIntervals(trustedIpRanges.split(","));
```

Comment:
Use COMMA_SPLITTER that's already defined rather than String.split(",")

Reply:
Done